server/methods/core/cart.js

```diff
@@ -880,15 +951,12 @@ Meteor.methods({
       }
     } else { // or if we remove address itself, when we run this part we assume
       // that the billing/shipping arrays can hold only one element [0]
-      if (cart.billing && typeof cart.billing[0].address === "object" &&
-        cart.billing[0].address._id === addressId) {
+      if (cart.billing && typeof cart.billing[0].address === "object" && cart.billing[0].address._id === addressId) {
```

Comment:
I didn't change this code, just removed a line break. But I don't think we want to change the behavior because we are removing this address from the cart not just setting/remove the default address.